src/components/appBar/CyVerseAppBar.js

```diff
@@ -102,10 +104,24 @@ function CyverseAppBar(props) {
     const router = useRouter();
     const { intl, children } = props;
 
+    const [userProfile, setUserProfile] = useUserProfile();
+
+    React.useEffect(() => {
+        const fetchUserProfile = async function() {
+            const profile = await fetch("/api/profile", {
+                method: "GET",
+                credentials: "include",
+            })
+                .then((resp) => resp.json())
+                .catch((e) => console.log(`error ${e.message}`));
+            setUserProfile(profile);
+        };
+        fetchUserProfile(setUserProfile);
+    }, [setUserProfile]);
```

Comment:
I'll have to read up on contexts more because `setUserProfile` is a useEffect dependency I did not expect that I find interesting 🤔 
There's also the `callApi` function in the `common` folder now if you wanted to use that for `fetch`, but since your usage is so small, it doesn't really matter.

Reply:
It turned out to be easiest to make the API call from within a React component (because `fetch` wasn't available otherwise), so I ended up not putting the call inside the context. `callApi` sounds really useful. I'll probably update the code to use it instead.

Reply:
I was getting warnings when I didn't pass in `setUserProfile` as a dependency. My understanding is that because it's available outside of the `useEffect` hook, it has to be listed as a dependency. Otherwise, if `setUserProfile` changed somehow, the user profile wouldn't be retrieved again.

`setUserProfile` doesn't need to be passed to setUserProfile. I was originally planning on moving `fetchUserProfile` to another file but then decided against it. I removed the parameter from the function definition, but forgot to remove it from the function call. 😖 Thanks for catching that.